inference-engine/src/mkldnn_plugin/nodes/mkldnn_gemm_node.cpp

```diff
@@ -187,17 +187,44 @@ void MKLDNNGemmNode::createPrimitive() {
     }
 }
 
-void MKLDNNGemmNode::execute(mkldnn::stream strm) {
+inline void process_gemm(char transa, char transb, int M, int N, int K, float alpha, const float *A, int lda,
+                         const float *B, int ldb, float beta, float *C, int ldc) {
+    mkldnn_sgemm(transa, transb, M, N, K, alpha, A, lda, B, ldb, beta, C, ldc);
+}
+
+inline void process_gemm(char transa, char transb, int M, int N, int K, float alpha, const uint8_t *A, int lda,
+                         const int8_t *B, int ldb, float beta, float *C, int ldc) {
+    const int32_t co = 0;
+    int32_t *Ci = reinterpret_cast<int32_t *>(C);
+    mkldnn_gemm_u8s8s32(transa, transb, 'F', M, N, K, alpha, A, lda, 0, B, ldb, 0, beta, Ci, ldc, &co);
+    parallel_for(M * N, [&](size_t i) {
+        C[i] = Ci[i];
+    });
```

Comment:
Current precision flow in NG and LPT assumes FP32 as Gemm output precision. We can get U8/I8 output only in case of fused FQ layer (which is re-quantize in fact), but now this fusing is not supported.
In general I agree with you. It looks better to keep node semantic with I32 and to allow MKLDNNGraph deal with data conversions. From the other side right now we are doing data conversions "inplace" (inside minibatch cycle) which should be more effective from data locality standpoint than explicit reorder. 
Given all the statements above I would suggest to keep all the things as is if you don't mind.

Reply:
Ok. Agree.